### PR TITLE
Better Release Slack Notifications

### LINKED ISSUE #44154: Automatically send useful release slack messages
Update in the same slack thread:
- pre-release test result (success or failure)
- publish start message
- release complete message

### PR DESCRIPTION
**Pull request by iethree**
closes https://github.com/metabase/metabase/issues/44154

### Description

Adds additional automated slack messages to the release process, and updates an existing message
- Build start message now tags the person who triggered the build
- Adds a message to notify of the result of pre-release testing
- Adds a message to notify that the publish job has started
- Automates (the previously manual) step of notifying that the publish is complete.
- Automates (the previously manual) step of notifying the #general channel that the release is complete

Screen Shot 2024-06-14 at 10 59 45 AM

✅ Test run:
- [tag workflow](https://github.com/iethree/metabase/actions/runs/9513241398)
- [test workflow](https://github.com/iethree/metabase/actions/runs/9513263640)
- [publish workflow](https://github.com/iethree/metabase/actions/runs/9513357424)
- [slack thread](https://metaboat.slack.com/archives/C01BWAZJE0N/p1718354425203489)

### CODE REVIEW

File: release/src/slack.ts [hunk 1]
```diff
@@ -148,21 +152,59 @@ export async function sendPreReleaseStatus({
...
+async function sendSlackReply({ channelName, message, messageId, broadcast }: {channelName: string, message: string, messageId?: string, broadcast?: boolean}) {
+  const channelId = await getSlackChannelId(channelName);
+  if (!channelId) {
+    throw new Error(`Could not find channel ${channelName}`);
+  }
+
+  return slack.chat.postMessage({
+    channel: channelId,
+    text: message,
+    thread_ts: messageId, // if this is empty it should post in the channel
```
Comment (crisptrutski) [inline release/src/slack.ts:203]: Nifty

File: release/src/slack.ts [hunk 2]
```diff
@@ -182,58 +224,124 @@ function githubRunLink(
...
+  );
+
+  const githubBuildLink = githubRunLink("_🏗️ CI Build_", runId, owner, repo);
+
+  const preReleaseMessage = [
+    releaseCommitLink,
+    milestoneLink,
+    githubBuildLink,
+    userName ? `started by ${mentionUserByGithubLogin(userName)}` : null
+  ].filter(Boolean).join(" - ");
```
Comment (crisptrutski) [inline release/src/slack.ts:267]: Surprised this doesn't need line breaks 😕

File: release/src/slack.ts [hunk 3]
```diff
@@ -182,58 +224,124 @@ function githubRunLink(
...
-  stage: BuildStage;
   version: string;
   runId: string;
   releaseSha: string;
+  channelName: string,
+  userName: string,
 }) {
-
   const title = getReleaseTitle(version);
   const space = "\n";
```
Comment (crisptrutski) [inline release/src/slack.ts:244]: Surely this should be called `linebreak`?

**Review by crisptrutski — APPROVED** [review]
What a gift to both doers and watchers 💌

File: release/src/slack.ts (hunk 3 above)
Comment (iethree) [reply release/src/slack.ts:244]: 😅

**Pull request merged**